dtdl-parser/src/object_info_impl.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+use serde_json::Value;
+use std::any::Any;
+use std::collections::HashMap;
+
+use crate::complex_schema_info::ComplexSchemaInfo;
+use crate::dtmi::Dtmi;
+use crate::entity_info::EntityInfo;
+use crate::entity_kind::EntityKind;
+use crate::field_info::FieldInfo;
+use crate::object_info::ObjectInfo;
+use crate::schema_info::SchemaInfo;
+
+pub struct ObjectInfoImpl {
```

Comment:
here and for other similar structs and impls - why not https://docs.rs/readonly/latest/readonly/? Then there will be no need for methods to simply return fields.

Reply:
Because I need to use traits.  Happy to discuss this with you.

Reply:
Because I need to use traits.